src/iosource/Manager.h

```diff
@@ -110,22 +119,57 @@ class Manager
 	 *
 	 * @param ready A vector used to return the set of sources that are ready.
 	 */
-	void FindReadySources(std::vector<IOSource*>* ready);
+	void FindReadySources(ReadySources* ready);
 
 	/**
 	 * Registers a file descriptor and associated IOSource with the manager
-	 * to be checked during FindReadySources.
+	 * to be checked during FindReadySources. This will register the file
+	 * descriptor to check for read events.
 	 *
 	 * @param fd A file descriptor pointing at some resource that should be
 	 * checked for readiness.
 	 * @param src The IOSource that owns the file descriptor.
 	 */
-	bool RegisterFd(int fd, IOSource* src);
+	[[deprecated("Remove in v5.1. Use RegisterReadFd instead.")]] bool RegisterFd(int fd,
```

Comment:
As in chat. I'm quite interested in that change for ZeekJS because it registers the Node.js's uv_loop FD with Zeek and I hadn't been aware Zeek only monitors for read events (and that may explain some unusual behavior observed).

I've been wondering if the existing `RegisterFD()` could be extended with a `mode`,  `filter` or `type` argument that defaults to `Read`, but users can then pass `Read|Write` (or something appropriate for C++).

Might be nicer as a user and also avoid the deprecation unless I'm glossing over something?

```
bool RegisterFd(int fd, IOSource *src, int filter=READ);
```

Reply:
That seems reasonable. Let me fix that.

Reply:
Alright, see if that looks better. I kept the four new methods but made them private, and reused the READ/WRITE flags from IOSource as the argument to `RegisterFd`/`UnregisterFd`.

Reply:
Thanks! I had thought it might be nice to be able to register in one call rather than calling it twice.
```
int loop_fd = ...
zeek::iosource_mgr->RegisterFd(loop_fd, loop_io_source, IoSource::READ|IoSource::WRITE);
// or because | on enum doesn't work...
zeek::iosource_mgr->RegisterFd(loop_fd, loop_io_source, IoSource::READ_WRITE)
```
...but doing two calls certainly works and don't intend to hold up the important work in this MR (and maybe could be tweak it later on, too).

Reply:
Is https://github.com/zeek/zeek/pull/1984/commits/531eba7c892599d338c13ac82cee87a34c1f775b more what you were thinking? I realized that splitting it into two methods was causing it to have to wake up more often than it should if you had to call the method twice.

Reply:
Thank you.